### ISSUE TITLE
Installer can't see HP Disk controller

### ISSUE THREAD

**Issue by the-maldridge** [voidlinux/void-mklive#44]
During a recent install attempt, the installer couldn't see the raid controller, nor could cfdisk.  The disks did show up in /dev/ though and were mountable.  The hardware in question is an HP GL360 5th generation server.  See below for an lspci of the hardware on the machine:

```
00:00.0 Host bridge: Intel Corporation 5000P Chipset Memory Controller Hub (rev b1)
00:02.0 PCI bridge: Intel Corporation 5000 Series Chipset PCI Express x4 Port 2 (rev b1)
00:03.0 PCI bridge: Intel Corporation 5000 Series Chipset PCI Express x4 Port 3 (rev b1)
00:04.0 PCI bridge: Intel Corporation 5000 Series Chipset PCI Express x8 Port 4-5 (rev b1)
00:05.0 PCI bridge: Intel Corporation 5000 Series Chipset PCI Express x4 Port 5 (rev b1)
00:06.0 PCI bridge: Intel Corporation 5000 Series Chipset PCI Express x4 Port 6 (rev b1)
00:07.0 PCI bridge: Intel Corporation 5000 Series Chipset PCI Express x4 Port 7 (rev b1)
00:10.0 Host bridge: Intel Corporation 5000 Series Chipset FSB Registers (rev b1)
00:10.1 Host bridge: Intel Corporation 5000 Series Chipset FSB Registers (rev b1)
00:10.2 Host bridge: Intel Corporation 5000 Series Chipset FSB Registers (rev b1)
00:11.0 Host bridge: Intel Corporation 5000 Series Chipset Reserved Registers (rev b1)
00:13.0 Host bridge: Intel Corporation 5000 Series Chipset Reserved Registers (rev b1)
00:15.0 Host bridge: Intel Corporation 5000 Series Chipset FBD Registers (rev b1)
00:16.0 Host bridge: Intel Corporation 5000 Series Chipset FBD Registers (rev b1)
00:1d.0 USB controller: Intel Corporation 631xESB/632xESB/3100 Chipset UHCI USB Controller #1 (rev 09)
00:1d.1 USB controller: Intel Corporation 631xESB/632xESB/3100 Chipset UHCI USB Controller #2 (rev 09)
00:1d.2 USB controller: Intel Corporation 631xESB/632xESB/3100 Chipset UHCI USB Controller #3 (rev 09)
00:1d.3 USB controller: Intel Corporation 631xESB/632xESB/3100 Chipset UHCI USB Controller #4 (rev 09)
00:1d.7 USB controller: Intel Corporation 631xESB/632xESB/3100 Chipset EHCI USB2 Controller (rev 09)
00:1e.0 PCI bridge: Intel Corporation 82801 PCI Bridge (rev d9)
00:1f.0 ISA bridge: Intel Corporation 631xESB/632xESB/3100 Chipset LPC Interface Controller (rev 09)
00:1f.1 IDE interface: Intel Corporation 631xESB/632xESB IDE Controller (rev 09)
01:03.0 VGA compatible controller: Advanced Micro Devices [AMD] nee ATI ES1000 (rev 02)
01:04.0 System peripheral: Compaq Computer Corporation Integrated Lights Out Controller (rev 03)
01:04.2 System peripheral: Compaq Computer Corporation Integrated Lights Out  Processor (rev 03)
01:04.4 USB controller: Hewlett-Packard Company Integrated Lights-Out Standard Virtual USB Controller
01:04.6 IPMI SMIC interface: Hewlett-Packard Company Integrated Lights-Out Standard KCS Interface
02:00.0 PCI bridge: Broadcom EPB PCI-Express to PCI-X Bridge (rev c3)
03:00.0 Ethernet controller: Broadcom Corporation NetXtreme II BCM5708 Gigabit Ethernet (rev 12)
04:00.0 PCI bridge: Broadcom EPB PCI-Express to PCI-X Bridge (rev c3)
05:00.0 Ethernet controller: Broadcom Corporation NetXtreme II BCM5708 Gigabit Ethernet (rev 12)
06:00.0 RAID bus controller: Hewlett-Packard Company Smart Array Controller (rev 01)
09:00.0 PCI bridge: Intel Corporation 6311ESB/6321ESB PCI Express Upstream Port (rev 01)
09:00.3 PCI bridge: Intel Corporation 6311ESB/6321ESB PCI Express to PCI-X Bridge (rev 01)
0a:00.0 PCI bridge: Intel Corporation 6311ESB/6321ESB PCI Express Downstream Port E1 (rev 01)
0a:01.0 PCI bridge: Intel Corporation 6311ESB/6321ESB PCI Express Downstream Port E2 (rev 01)
0a:02.0 PCI bridge: Intel Corporation 6311ESB/6321ESB PCI Express Downstream Port E3 (rev 01)
```

**Comment by bougyman** [comment]
http://pastebin.com/sniGjMmZ probably doesn't see cciss as a disk. Will investigate.

**Comment by the-maldridge** [comment]
Per request, here is the output of tree /sys/block:

```
/sys/block/
├── cciss!c0d0 -> ../devices/pci0000:00/0000:00:03.0/0000:06:00.0/cciss0/c0d0/block/cciss!c0d0
├── loop0 -> ../devices/virtual/block/loop0
├── loop1 -> ../devices/virtual/block/loop1
├── loop2 -> ../devices/virtual/block/loop2
├── loop3 -> ../devices/virtual/block/loop3
├── loop4 -> ../devices/virtual/block/loop4
├── loop5 -> ../devices/virtual/block/loop5
├── loop6 -> ../devices/virtual/block/loop6
├── loop7 -> ../devices/virtual/block/loop7
└── sr0 -> ../devices/pci0000:00/0000:00:1f.1/host0/target0:0:0/0:0:0:0/block/sr0

10 directories, 0 files
```

and of /dev/block

```
/dev/block/
├── 104:0 -> ../cciss/c0d0
├── 104:1 -> ../cciss/c0d0p1
├── 104:2 -> ../cciss/c0d0p2
├── 104:5 -> ../cciss/c0d0p5
├── 11:0 -> ../sr0
├── 7:0 -> ../loop0
├── 7:1 -> ../loop1
├── 7:2 -> ../loop2
├── 7:3 -> ../loop3
├── 7:4 -> ../loop4
├── 7:5 -> ../loop5
├── 7:6 -> ../loop6
└── 7:7 -> ../loop7

0 directories, 13 files
```

**Comment by the-maldridge** [comment]
This is the output of ls -l /dev/cciss

```
total 0
brw-rw---T 1 root disk 104, 0 Oct  9 23:38 c0d0
brw-rw---T 1 root disk 104, 1 Oct  9 23:38 c0d0p1
brw-rw---T 1 root disk 104, 2 Oct  9 23:38 c0d0p2
brw-rw---T 1 root disk 104, 5 Oct  9 23:38 c0d0p5
```

**Comment by xtraeme** [comment]
Please try this patch: https://gist.github.com/anonymous/22c70c130896ae2c90a1

**Comment by the-maldridge** [comment]
Installer encountered an issue wherein it was able to correctly partition the disks, I was able to assign the resulting partitions to mountpoints, but the installer would not install.  The error text from https://github.com/voidlinux/void-mklive/blob/8261f90a2aaee2d16d0d719d8656a82e7239beb0/installer.sh.in#L651 was displayed and the installer would not proceed.  Manually changing that check to always report true did not result in success.  I'm happy to continue testing but cannot take the system offline to do a full wipe again.

**Comment by xtraeme** [comment]
Fixed with 7c699d32c2f0b97a65f423f219e15d09c7d88003 and c0e16252c3ded26bd63db539b0f677599d671620

Verified by @the-maldridge 

Thanks!
